### PR TITLE
Added Help Command

### PR DESCRIPTION
**Pull request by bdbch**

### CODE REVIEW

File: lib/actions/help.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const commands = require('../../config/commands.json')
```
Comment (bdbch) [inline lib/actions/help.js:1]: Maybe we can do this better later and somehow define this in the js files so the help list is created automatically

File: lib/actions/help.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+function help (bot, options) {
+  // When discord emits 'ready'.
+  // Used to handle things outside
+  // of the running function.
+
+  return function run (message, args) {
+    // When this command gets triggered.
+    let helpMessage = 'Here are our commands: \n\n'
+    for(i = 0; i < commands.length; i++) {
```
Comment (Dragory) [inline lib/actions/help.js]: `i` probably shouldn't be a global variable; `let i` should be fine

**Comment by jamen** [comment]
Each command will provide multiple propreties to describe the name, usage, description, etc. (whatever we want) using syntax like this:

``` js
foo.command = 'foo';
foo.usage = 'foo <blah>';
// ...

function foo (bot, options) {
  // ...
}
```

We should use this information instead of a separate config, because then you can also add help menus for 3rd party actions as well.

File: lib/actions/help.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      helpMessage += "**" + commands[i].name + '**\n'
+      for(o = 0; o < commands[i].commands.length; o++) {
```
Comment (Dragory) [inline lib/actions/help.js]: Same as above here

**Review by Dragory — CHANGES_REQUESTED** [review]

**Comment by jamen** [comment]
Also, a note on the configurations, you should be using:

``` js
function (bot, options) {
  // ...
```

The `options` parameter contains the configuration provided by [`rc`](https://npmjs.org/rc), you should also utility functions inside of there to have access to those.

**Comment by bdbch** [comment]
@jamen as I said, never worked with this system. Could you provide a demo on how to work with the function wrapping the run function?

I'll update this.

**Comment by jamen** [comment]
I'll send a pull request updating `role`, after I do a pull request updating docs on configuration (what is on my current working branch)

**Comment by Dragory** [comment]
If the names/descriptions were part of the commands itself, maybe this would be a good reason to switch to using ES6 classes for the commands? Have the name etc. as its properties.

**Comment by jamen** [comment]
ES6 classes don't have defined properties, only methods, and properties set inside the constructor.  They're sugar over constructor + prototypes.  We also want to keep a functional structure.  I don't see anything wrong with the system.  Fast and clean.  :)

**Comment by bdbch** [comment]
@jamen you could still use getters and setters for properties, but still I don't mind as long as I can understand the concept later + it works haha.

**Comment by jamen** [comment]
Merge with master.  I've changed actions folder from `cordlr/lib/actions` to `cordlr/actions`.

Going to start on `role.js`'s style now.

**Comment by jamen** [comment]
Updated `roles.js`.  See that and try to mimic the style.  Let me know if you have any questions.

**Comment by bdbch** [comment]
I'll update this branch as soon as your role-cleanup branch is merged.

**Pull request closed without merging**